chia/types/blockchain_format/program.py

```diff
@@ -133,6 +133,8 @@ def run(self, args) -> "Program":
     def curry(self, *args) -> "Program":
         fixed_args: Any = 1
         for arg in reversed(args):
+            if type(arg) == SerializedProgram:
```

Comment:
Is this really necessary? As long as both `Program` and `SerializedProgram` descend from `CLVMObject`, they should interoperate.

Reply:
I see, `SerializedProgram` is a base class. What happens if you make it a subclass of `CLVMObject` (or even `Program`)? You'd have to override `.atom` and `.pair` (decorate with `@property def atom(self): ...`).

Reply:
Check for `0xff` as the first byte to determine `atom` vs `pair`. Then call `deserialized` and populate `._atom` or `._pair` so you only have to deserialize once (this is similar to the strategy I'm trying in my rewrite).

Reply:
right, `SerializedProgram` doesn't implement `CLVMObject` or derive from `SExp`. I would expect it would be more complicated to implement `CLVMObject` than adding this special case.